File: packages/composition/test/mock.ts
import { faker } from "@faker-js/faker";

export const mock = {
  emailAddress: faker.internet.email(),
  firstName: faker.name.firstName(),
  lastName: faker.name.lastName(),
  password: faker.internet.password(),
  title: faker.commerce.productName(),
  imageUrls: [
    `${faker.image.imageUrl()}/${faker.random.alphaNumeric(15)}`,
    `${faker.image.imageUrl()}/${faker.random.alphaNumeric(15)}`,
  ],
  description: faker.lorem.paragraphs(),
  startDate: faker.date.past().toISOString(),
  endDate: faker.date.soon().toISOString(),
  location: {
    address: faker.address.streetAddress(),
    city: faker.address.cityName(),
    country: faker.address.country(),
    countryCode: faker.address.countryCode(),
    phone: faker.phone.number("##########"),
    province: "British Columbia",
    provinceCode: "BC",
    zip: faker.address.zipCode(),
    coordinate: {
      latitude: Number(faker.address.latitude()),
      longitude: Number(faker.address.longitude()),
    },
  },
};

File: packages/service-space/test/data/mock.go
package data

import (
	"github.com/bxcodec/faker/v3"
	"github.com/kokiebisu/rental-storage/service-space/internal/core/domain/space"
	"github.com/kokiebisu/rental-storage/service-space/internal/core/domain/space/coordinate"
	location "github.com/kokiebisu/rental-storage/service-space/internal/core/domain/space/location"
)

var (
	MockSpace = space.DTO{
		UId:         faker.UUIDDigit(),
		Title:       "A beautiful garage",
		Description: "BLAH BLAH",
		LenderId:    faker.UUIDDigit(),
		Location: location.DTO{
			Address:      "Blah Blah",
			City:         "Vancouver",
			Country:      "Canada",
			CountryCode:  "CA",
			Phone:        "123-456-7891",
			Province:     "British Columbia",
			ProvinceCode: "BC",
			Zip:          "V193XA",
			Coordinate: coordinate.DTO{
				Latitude:  faker.Latitude(),
				Longitude: faker.Longitude(),
			},
		},
		ImageUrls: []string{
			faker.URL(),
			faker.URL(),
		},
		CreatedAt: faker.Date(),
		UpdatedAt: faker.Date(),
	}
)
